self_hosting_machinery/embeddings/inference_embed.py

```diff
@@ -0,0 +1,108 @@
+import sys
+import logging
+import time
+import signal
+import socket
+
+from refact_scratchpads_no_gpu.stream_results import infserver_session
+from refact_scratchpads_no_gpu.stream_results import validate_description_dict
+from refact_scratchpads_no_gpu.stream_results import UploadProxy
+from refact_scratchpads_no_gpu.stream_results import completions_wait_batch
+
+from self_hosting_machinery.embeddings.encoder import RefactEncoder
+
+quit_flag = False
+log = logging.getLogger("MODEL").info
+
+
+def worker_loop(
+        model_name: str,
+        compile_only: bool = False,
+):
+    log("STATUS loading model")
+
+    enc_model = RefactEncoder()
+
+    class DummyUploadProxy:
+        def upload_result(*args, **kwargs):
+            pass
+
+        def check_cancelled(*args, **kwargs):
+            return set()
+
+    dummy_calls = [
+        {
+            'id': 'emb-legit-42',
+            'function': 'embeddings',
+            'inputs': "Common Knowledge",
+            'created': time.time(),
+        }
+    ]
+    log("STATUS test batch")
+    enc_model.infer(dummy_calls[0], DummyUploadProxy, {}, log)
+    if compile_only:
+        return
+
+    req_session = infserver_session()
+    description_dict = validate_description_dict(
+        f'{model_name}_{socket.getfqdn()}',
+        "account_name",
+        model=model_name, B=1, max_thinking_time=10,
+    )
+    upload_proxy = UploadProxy(upload_q=None, cancelled_q=None)
+    upload_proxy.start_upload_result_daemon()
+
+    while not quit_flag:
+        upload_proxy.keepalive()
+        upload_proxy.cancelled_reset()
+        retcode, request_batch = completions_wait_batch(
+            req_session, description_dict, verbose=False)
+        ts_arrived = time.time()
+        if retcode == "OK":
+            for request in request_batch:
+                upload_proxy_args = {
+                    "description_dict": description_dict,
+                    "original_batch": [request],
+                    "idx_updated": [0],
+                    "tokens": None,
+                    "ts_arrived": ts_arrived,
+                    "ts_batch_started": time.time(),
+                    "ts_prompt": 0,
+                    "ts_first_token": 0,
+                    "ts_batch_finished": 0,
+                }
+                enc_model.infer(request, upload_proxy, upload_proxy_args, log)
+        elif retcode == "WAIT":
+            pass
+        else:
+            time.sleep(10)
+
+    upload_proxy.stop()
+    log("clean shutdown")
+
+
+def catch_sigkill(signum, frame):
+    sys.stderr.write("caught SIGUSR1\n")
+    sys.stderr.flush()
+    global quit_flag
+    quit_flag = True
+
+
+if __name__ == "__main__":
```

Comment:
congrats, you implemented inference worker
I believe that there is a way to implement worker for embedding models